### PR TITLE
Problem: C++ version not correctly estimated

### PR DESCRIPTION
**Pull request by gummif**
Solution: Include headers before checking macros and add additional checks for _MSVC_LANG.

Fixes https://github.com/zeromq/cppzmq/issues/426

### CODE REVIEW

File: zmq.hpp [hunk 1]
```diff
@@ -32,20 +32,34 @@
 #endif
 #endif
 
+// included here for _HAS_CXX* macros
+#include <zmq.h>
+
+#if defined(_MSCV_LANG)
```
Comment (sigiesec) [inline zmq.hpp:38]: I guess this should be `_MSVC_LANG`? Also in the line below.

File: zmq.hpp [hunk 2]
```diff
@@ -32,20 +32,34 @@
...
+#define CPPZMQ_LANG _MSCV_LANG
+#elif defined(__cplusplus)
+#define CPPZMQ_LANG __cplusplus
+#else
+#define CPPZMQ_LANG 0
```
Comment (sigiesec) [inline zmq.hpp:43]: I think this should rather be some `#error`. If `__cplusplus` is not defined, we are not compiling in C++ mode, and that can't work.

**Review by sigiesec — CHANGES_REQUESTED** [review]
Sorry for taking so long to look at this.

Is there a way we could test this is working as expected? In particular with MSVC? We can merge this independently of addressing that, but it would be generally nice, as it might easily break. (Obviously it doesn't work as of now as the define used by MSVC is misspelled, still all jobs seem fine)

File: zmq.hpp (hunk 1 above)
Comment (gummif) [reply zmq.hpp:38]: Yes, good catch.

**Comment by gummif** [comment]
This breaks VS 2015 for some reason. Will take a look later.

**Pull request merged**